src/app/components/signer/signer.component.ts

```diff
@@ -80,23 +81,36 @@ export class SignerComponent implements OnInit {
   }
 
   signData() {
-    this.dialogsService.open(PasswordDialogComponent, ModalSizeTypes.ExtraExtraLarge, this.dialogMessage)
+    let dialogMessage = 'Enter your vault password to sign the ';
+    dialogMessage += this.requestType == RequestType.Basic || this.requestType == 'data'
+      ? 'data'
+      : 'transaction';
+
+    this.dialogsService.open(PasswordDialogComponent, ModalSizeTypes.ExtraExtraLarge, dialogMessage)
       .subscribe((vaultPassword: string) => {
         if (vaultPassword) {
           this.spinner.show();
 
-          if (this.requestType == RequestType.Basic) {
-            const dataToSign = typeof this.request.data == "string"
-              ? this.request.data
-              : JSON.stringify(this.request.data);
+          if (this.requestType == RequestType.Basic || this.requestType == "data") {
+            let data;
+            if (this.requestType == "data") {
+              data = this.request.requestInfo.data;
+            } else {
+              data = this.request.data;
+            }
+
+            const dataToSign = typeof data == "string"
+              ? data
+              : JSON.stringify(data);
 
             const signingKeyOrAddress = this.selectedKeyId
               ? this.availableKeys.find(dk => dk.id == this.selectedKeyId)
               : this.selectedFactomAddress;
 
             this.signBasicRequest(dataToSign, signingKeyOrAddress, vaultPassword);
           } else {
-            const dataToSign = Buffer.from(Object.values(this.request.data));
+            const dataToSign = this.buildTransactionDataToSign();
```

Comment:
I'm not sure how my tests were passing the other day, but I now realized that the way we've coded the signing of PegNet requests right now does not work.

The reason is that at the moment we only return the signature for a conversion or a transfer back to the wallet and leave it to wallet to construct the actual entry that will needs to be recorded on-chain. However, the wallet cannot do this, because it doesn't know the timestamp that was used to generate the signature and that timestamp is one of the `ExtID`s that has to be included in the Factom entry.

I see two options to fix this:

1. For PegNet requests, in the `message` field of the `SigningResponse` we return the complete entry that needs to be recorded on-chain (i.e. the ExtIDs and the content). We can still have the `signature` field be the same.
1. For PegNet requests, we change the format of the data returned in the `SigningResponse` and drop the `signature` & `message` fields, instead including a new field `entry`, which contains the ExtIDs and the content of the entry.

I'm leaning towards the second option because I think it's cleaner, but let's discuss tomorrow.